anneal.go

```diff
@@ -45,6 +45,10 @@ type SASettings struct {
 	// during the last third of iterations
 	AnnealingFactor float64
 	Settings
+	// Random generator, overriding math/rand's default rand.Source when specified.
+	// This is in particular useful for tests predictibility, as since Go 1.24,
+	// rand.Seed() is a no-op; but still as effective with non-default Sources.
+	*rand.Rand
```

Comment:
you have a Settings here, so you have Settings.Rand ? as you defined Rand in Settings 

no ?